tests/unit/Collectors/ScheduledTaskCollectorTest.php

```diff
@@ -21,27 +34,163 @@ class ScheduledTaskCollectorTest extends Unit
+        $this->configMock = Mockery::mock(Config::class);
+        $this->eventMock = Mockery::mock(Event::class);
+        $this->eventClockMock = Mockery::mock(EventClock::class);
+
+        $this->eventMock->command = self::COMMAND_NAME;
+        $this->eventMock->exitCode = 0;
+
         $eventCounter = new EventCounter();
-        $eventClock = new EventClock();
 
         $this->collector = new ScheduledTaskCollector(
             $this->app,
-            new Config([]),
+            $this->configMock,
             new RequestStartTime(0.0),
             $eventCounter,
-            $eventClock
+            $this->eventClockMock
         );
 
-        $this->collector->useAgent(Mockery::mock(Agent::class));
+        $this->collector->useAgent($this->agentMock);
+    }
+
+    protected function _after(): void
+    {
+        $this->dispatcher->forget(ScheduledTaskStarting::class);
+        $this->dispatcher->forget(ScheduledTaskSkipped::class);
+        $this->dispatcher->forget(ScheduledTaskFinished::class);
+    }
+
+    protected function patternConfigReturn($configIgnore = null): void
+    {
+        $this->configMock->expects('get')
+            ->with('elastic-apm-laravel.transactions.ignorePatterns')
+            ->andReturn($configIgnore);
     }
 
     public function testCollectorName(): void
     {
         self::assertEquals('scheduled-task-collector', $this->collector->getName());
     }
+
+    public function testScheduledTaskStartingListenerIgnored(): void
+    {
+        $this->eventMock->command = 'work:do';
+        $this->patternConfigReturn(self::TASK_IGNORE_PATTERN);
+        $this->agentMock->shouldNotReceive(['startTransaction', 'getTransaction']);
+
+        $this->dispatcher->dispatch(new ScheduledTaskStarting($this->eventMock));
+    }
+
+    public function testScheduledTaskSkippedListenerIgnored(): void
+    {
+        $this->eventMock->command = 'work:do';
+        $this->patternConfigReturn(self::TASK_IGNORE_PATTERN);
+        $this->agentMock->shouldNotReceive(['startTransaction', 'getTransaction']);
+
+        $this->dispatcher->dispatch(new ScheduledTaskSkipped($this->eventMock));
+    }
+
+    public function testScheduledTaskFinishedListenerIgnored(): void
+    {
+        $this->eventMock->command = 'work:do';
+        $this->patternConfigReturn(self::TASK_IGNORE_PATTERN);
+        $this->agentMock->shouldNotReceive(['startTransaction', 'getTransaction']);
+
+        $this->dispatcher->dispatch(new ScheduledTaskFinished($this->eventMock, 1000.0));
+    }
+
+    public function testScheduledTaskStartingListener(): void
+    {
+        $this->patternConfigReturn();
+
+        $this->eventClockMock->expects('microtime')->andReturn(1000);
+
+        $this->agentMock->expects('getTransaction')
+            ->with(self::COMMAND_NAME)
+            ->andReturn(null);
+
+        $this->agentMock->expects('startTransaction')
+            ->with(self::COMMAND_NAME, [], 1000)
```

Comment:
What is that `1000`? Start time, why 1000, I'm lost.

Reply:
It was like that in the command collector tests, C&P. Do you want me to create a constant?

Reply:
I guess it's fine -- technically it is 8 lines above. Just looks weird for some reason I can't explain.